packages/ui/src/components/QueryBuilder/QueryPills/BooleanQueryPill.tsx

```diff
@@ -19,43 +21,64 @@ interface IBooleanQueryPillProps {
     getColorForReference?: (refIndex: number) => string;
 }
 
-const isNotEnd = (props: IBooleanQueryPillProps, index: number) => props.query.content.length - 1 > index;
-
-const BooleanQueryPill = (props: IBooleanQueryPillProps) => (
-    <Fragment>
-        {props.query.content.map((f: any, i: number) => (
-            <Space key={i} size={0} style={{ padding: '2px 0px' }}>
-                {isBooleanOperator(f) ? (
-                    <BooleanQueryPill {...props} query={f} />
-                ) : isReference(f) ? (
-                    <ReferenceQueryPill
-                        isBarActive={props.isActive}
-                        refIndex={f as number}
-                        onRemove={() => props.onRemoveReference(f as number, props.query)}
-                        getColorForReference={props.getColorForReference}
-                    />
-                ) : isSet(f) ? (
-                    <SetQueryPill
-                        isBarActive={props.isActive}
-                        valueFilter={f as IValueFilter}
-                        onRemove={() => props.onRemoveReference(f as number, props.query)}
-                    />
-                ) : (
-                    <FieldQueryPill
-                        isBarActive={props.isActive}
-                        onRemove={() => props.onRemoveFacet(f, props.query)}
-                        valueFilter={f as IValueFilter}
-                    />
-                )}
-                {isNotEnd(props, i) && (
-                    <Combiner
-                        onChange={(type) => props.onCombineChange!(props.parentQueryId, type)}
-                        type={props.query.op}
-                    />
-                )}
-            </Space>
-        ))}
-    </Fragment>
-);
+const isNotEnd = (content: any[], index: number) => content.length - 1 > index;
+
+const BooleanQueryPill = (props: IBooleanQueryPillProps) => {
+    const newQueryDict: { [content: string]: any } = {};
+    props.query.content.map((f: any) => {
```

Comment:
Does a forEach could be more appropriate here?

[optional] I'm a big fan of the earlier return pattern

```
        if (newQueryDict.hasOwnProperty(f.content.field)) {
            newQueryDict[f.content.field] = cloneDeep(newQueryDict[f.content.field]);
            newQueryDict[f.content.field].content.value = newQueryDict[f.content.field].content.value.concat(
                f.content.value,
            );
            return;
        }
        
        newQueryDict[f.content.field] = cloneDeep(f);

```

Reply:
corrigé, en faite ca ne faisait pas de sense de prendre un map dans ce context. J'avais oublier de la modifier apres refacto